src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -140,19 +140,19 @@ def is_sku_available(cmd, sku_info, zone):
     is_available = True
     is_restrict_zone = False
     is_restrict_location = False
-    if not sku_info.restrictions:
+    if not sku_info.get('restrictions', []):
         return is_available
-    for restriction in sku_info.restrictions:
-        if restriction.reason_code == 'NotAvailableForSubscription':
+    for restriction in sku_info['restrictions']:
+        if restriction.get('reason_code', '') == 'NotAvailableForSubscription':
             # The attribute location_info is not supported in versions 2017-03-30 and earlier
             if cmd.supported_api_version(max_api='2017-03-30'):
                 is_available = False
                 break
-            if restriction.type == 'Zone' and not (
-                    set(sku_info.location_info[0].zones or []) - set(restriction.restriction_info.zones or [])):
+            if restriction['type'] == 'Zone' and not (
+                    set(sku_info['location_info'][0].get('zones', [])) - set(restriction['restriction_info'].get('zones', []))):
```

Comment:
There may be one case where "zones" is in the dictionary but the value is None
```suggestion
                    set(sku_info['location_info'][0].get('zones', []) or []) - set(restriction['restriction_info'].get('zones', []) or [])):
```